internal/assert/assert.go

```diff
@@ -4,11 +4,32 @@
 
 package assert
 
-import "testing"
+import (
+	"sort"
+	"strings"
+	"testing"
+)
 
 // Comparable asserts two comparable values that they equal.
 func Comparable[T comparable](t *testing.T, name string, want T, got T) {
 	if want != got {
 		t.Errorf(`%s: want "%v", got "%v"`, name, want, got)
 	}
 }
+
+// ElementsMatch asserts that two slices of strings contain the same values,
+// with no regard to the ordering.
+func ElementsMatch[S ~[]string](t *testing.T, want, got S) {
+	var wantClone S
+	copy(wantClone, want)
+	sort.Strings(wantClone)
+	var gotClone S
+	copy(gotClone, got)
```

Comment:
This doesn't work. The `copy` function only copies as many elements as fits, and a nil slice has a length of 0, meaning both of these `copy` operations result in two empty slices. In effect, this function doesn't assert anything :)

You need to initialize their sizes, such as like so:

```suggestion
	wantClone := make(S, len(want))
	copy(wantClone, want)
	sort.Strings(wantClone)
	gotClone := make(S, len(got))
	copy(gotClone, got)
```

Reply:
Aah, gotcha 👍🏻